### ISSUE TITLE
Periodic high RAM usage with Watferfox (tabs crashing sometimes)

### ISSUE THREAD

**Issue by Kherby** [BrowserWorks/waterfox#711]
Today i had 15 open Tabs at Watferfox (youtube, github and some forums) and i noticed that my System was running very slow.
I've looked into the Taskamager and saw that Waterfox was using over 9GB of my RAM (16GB total), a minute later the RAM usage was normal again (~1.3GB)
This also happend yesteday and RAM usage was even higher (~12GB) and my Notenbook was completly  running out of free RAM for a few moments and some tabs even crashed.

I never had this kind of an issue with Firefox ESR52 and i'm also using the same addons as with FF ESR 52.

Addons:
```
CanvasBlocker	0.5.2b	
Cookie Controller	6.1	
Disable WebRTC	1.0.20	
Enhancer for YouTube™	2.0.73	
HTTPS Everywhere	2018.6.21
NoScript	5.1.8.4	
Page Translate	1.6	
uBlock Origin	1.16.14

```
I'm going to do a clean install of Waterfox once again and hope that it will fix this Problem.
What can i do else? 

Btw, is there any way to completely remove everything of Waterfox out of my system so i can make sure that a clean installation is 100% clean and no old config files are beeing used?

 - OS: Win 10 x64
 - Version 1803
 - Waterfox 56.2.2 (64-bit)

**Comment by Ibuprophen** [comment]
Does it hang (not freeze) as if it was working really hard for some websites? 

Primarily for websites with allot of Flash, Java, and such within it? 

I just don't want to suggest something that may not apply to your issue without knowing this. 


~Ibuprophen

**Comment by Kherby** [comment]
Yes, Waterfox is really working hard at https://imgur.com/ and my TAB also crashed there. 
At the moment my TAB is crashing every time i access imgur.com and i also have to reload every other open TAB after the crash...

I've also forgot the screenshot at my initial post: https://imgur.com/jIuteMW

Could these crashes and my periodic high ram usage be related to the config changes i made to enable GPU video encoding and watch youtube 4k videos without having my CPU run at 70%?
See: https://github.com/MrAlex94/Waterfox/issues/706

My first experiences with Waterfox isn't really great so far... :(

**Comment by Ibuprophen** [comment]
That can sometimes happen if you have one or more outdated Adobe Shockwave, Adobe Flash Player and Java... 


~Ibuprophen

**Comment by Kherby** [comment]
I dont have java or adobe shockwave installed... Flashplayer comes with win 10 afaik.
I have no other additional browser plugins installed, only OpenH264-Videocodec which is preinstalled with Watefox.

I'm going to do a clean install of waterfox and it that doesn't help i have to switch back to FireFox ESR.

**Comment by Ibuprophen** [comment]
Windows 10 comes with the Flash Player but, it's only tied to the Explorer Browser and doesn't apply to 3rd party browsers like Waterfox and the ESR. 

I always have all 3 for the Waterfox Browsers because some websites also use Shockwave, some use Java and some use a combination of the 3.

I can only Suggest/Recommend... :-)


~Ibuprophen

**Comment by Kherby** [comment]
So i've reinstalled Waterfox once again and for now all seems to be good. I got 19 Tabs open at the moment, playing a 4k youtube video in the background and scrolling like a maniac trough imgur.com... I also have the same addons running as before... Hopefully it stays like that because i'm really happy with Waterfox if it doesn't freeze/crash and if it doesn't consume all my avalible RAM for no reason.
I'm going to let Waterfox run for the next hours and browse the internet from time to time. I'm really hoping that this memory leak won't return...

Btw, i've also installed the latest Intel drivers for my HD620 GPU (24.20.100.6194)... Maybe that helped at the end.
(fingers crossed that i don't have to return to Firefox ESR)

**Comment by Squall-Leonhart** [comment]
theres a weird issue that occurs with the parent process memory usage growing more and more any time a facebook message is recieved and the sound plays for it.

**Comment by Kherby** [comment]
High RAM usage is back again... I've no clue what's causing it.
It might be Malwarebytes Premium which also protects the waterfox.exe...

I guess that i have to switch back to FireFox ESR. :/

**Comment by Squall-Leonhart** [comment]
its not malware bytes

**Comment by grahamperrin** [comment]
> … Win … 

Actively check the file system. Does it appear to be OK? 

Maybe try an [Extension Conflict Troubleshooter](https://redd.it/93x02v) routine. Target the extension that's least likely to be troublesome, in your case maybe _Password Exporter_.  

For the high memory usage, I might suspect trouble with _HTTPS Everywhere_, however … 

> … At the moment my TAB is crashing every time i access imgur.com 

I don't imagine HTTPS Everywhere contributing to that. 

> and i also have to reload every other open TAB after the crash…

What exactly is wrong with a non-crashed tab if you do **not** reload it?

----

> … going to do a clean install of Waterfox once again …
> 
> … make sure that a clean installation is 100% clean 

You could use RegEdit after removing Waterfox, although I don't imagine it making a difference in this case. 

> and no old config files are beeing used? …

Refresh Waterfox. At about:support

> … Could these crashes and my periodic high ram usage be related to the config changes i made to enable GPU video encoding and watch youtube 4k videos without having my CPU run at 70%? …

If a change results in less use of CPU, then (loosely speaking) I don't imagine the change increasing the likelihood of a crash.

**Comment by WagnerGMD** [comment]
```
Try h264ify if YouTube videos stutter, take up too much CPU, eat battery life, or make your laptop hot.

By default, YouTube streams VP8/VP9 encoded video. However, this can cause problems with less powerful machines because VP8/VP9 is not typically hardware accelerated.

In contrast, H.264 is commonly hardware accelerated by GPUs, which usually means smoother video playback and reduced CPU usage. h264ify makes YouTube stream H.264 videos instead of VP8/VP9 videos.
```
It's just an idea because I'm wondering it might help for this kind of trouble ? You will find [the addon called h264ify](https://addons.mozilla.org/en-US/firefox/addon/h264ify/) there.

About the memory leak, it could be cause by YouTube specially when the video is very long (at least 2 hours or even more).
There are a few topics on this matter on the [Cyberfox Forum](https://8pecxstudios.com/Forums/viewforum.php?f=5) if you need another example(s).

**As reminder and warning : Don't forget to delete the profile !
Otherwise it won't be a clean (re)installation.**
```
## As one example, under Windows 10 (etc) you must remove the both folder :
C:\Users\WagnerGMD\AppData\Local\Waterfox
C:\Users\WagnerGMD\AppData\Roaming\Waterfox
```

**Comment by Kherby** [comment]
I've done a clean installation with deleting the profiles but it didn't help... 
I think there is something wrong with my System as other people doesn't seem to have this kind of problem.
Anyway thanks for the suggestion with h264ify!

**Comment by Squall-Leonhart** [comment]
Its not your system,  since waterfox 56.(maybe2) the browser operates correctly a few days and then begins to experience runaway leaks, I'm beginning to suspect service workers are involved.

**Comment by grahamperrin** [comment]
> … a few days and then begins to experience runaway leaks, …

Next time it happens, can you share a measurement? From about:memory

**Comment by Ibuprophen** [comment]
I'm just curious if these issues have any link to the App? 

Sometimes I'll have 2-3 +/- tabs open and, while doing various things, the app just closes on its own. 

When I reopen the app, I see an accurate history of the tabs that were open but, it seems like the issue was if I had completely stopped (Force Closed) the app when I didn't. It just happened when I was doing something that never happens when doing the same task. 

Also, it happens once every week to two weeks sporadically. 

I hope that I had explained this okay via text... 


~Ibuprophen

**Comment by Kherby** [comment]
@grahamperrin 
If it helps i can also share a measurement from about:memory.

**Comment by Kherby** [comment]
So I've started with a fresh install (also deleted the old profiles) and only installed ublock origin addon.
The high ram usage seems to be fixed for now so i guess it was related to one of my addons.
Is there any way to find out which addon what causing the issue? I'm kinda scared to install any new addon.

**Comment by laniakea64** [comment]
> Is there any way to find out which addon what causing the issue?

Add back your addons, one at a time, until you see the issue again.  Then disable all but the last-added addon to confirm that it's the culprit.

**Comment by Kherby** [comment]
After 2 days of usage i can definitly tell that my ram problem was related to one of my addons!! I have currently about 25 Tabs open and no problem with Waterfox...
I'll try to add back my old addons, one at a time, and hopefully i'll be able to find the evil one and let you guys know which one was causing the problem.
I really hope it's not noscript as i'm using this one pretty active!!

**Comment by Ibuprophen** [comment]
I remember looking into a similar situation that I had experienced a while ago and, I do agree with you @Kherby, that there's addons that can result heavily on a systems resources. 

It took me a while to locate them but, depending upon how many addons, it could potentially take a while to hone on those culprits. 

Mozilla should add a sort of "Resource" type of area/information about addons. A way to reflect a type of Low, Medium high resource usage indicator of sorts. 

There's even addons that may use a Low amount of Resources at first but, depending on what it does (over time) and such, can potentially end up using a High amount of Resources. 

Also, even though the App doesn't sync Addons directly back to the App itself, I just wanted to point out that addons that are added to the app itself will indeed sync somehow to the PC Browser. This does get very frustrating to keep an eye on. 



~Ibuprophen

**Comment by grahamperrin** [comment]
[Extension Conflict Troubleshooter : waterfox](https://www.reddit.com/r/waterfox/comments/93x02v/extension_conflict_troubleshooter/)

> definitly … related to one of my addons

OK, you can mark this issue as _closed_. Thanks for the follow-up.

**Comment by juneyourtech** [comment]
High RAM/CPU usage sometimes be can be caused by websites that use wayward CSS, JavaScript, or fonts. I've seen that with older versions of Firefox, and then I'd have investigate and test, and then to add lines to NoScript's ABE section to block the wayward file.

Update: If memory serves me right, then it was in GNU IceCat lists, that identified the possible culprit(s) as minified JavaScript and CSS.

**Comment by Kherby** [comment]
I still haven't found out which addon was causing the high ram usage...

**Comment by Kherby** [comment]
As far as i can tell the problem was related to the "Enhancer for YouTube™ addon"...
Very sad because i really like this addon but it seems that it's not working well with waterfox.

**Comment by grahamperrin** [comment]
### Enhancer for YouTube™

Around a year ago, before I began using Waterfox, I found performance (but not memory) problems with version 2.0.43. https://discourse.mozilla.org/t/-/20463?u=grahamperrin – I should add something there (when I last tested, earlier this year, I couldn't reproduce the problem … can't recall whether the test was with Waterfox or Firefox).

**Comment by BugZappr** [comment]
I may be a necroposting fool, but a while back, I came across a post claiming CanvasBlocker was causing a high CPU usage.  I've got about 2000 tabs in one session (not one window), in Tree Style Tabs, and I kept having a thing where if I went to certain sites, my memory usage for Firefox would go from about 2 GB to 8 GB after 1-2 mins., and I only have 16 GB installed.  At that point, not only would FF go unresponsive, but the entire system.  If I tried to start Task Manager, it would take a very long time to come up, but if I ended one of its subprocesses, everything would be back to normal.  Having remembered I'd had similar CPU usage probs. with it in the past, I disabled CanvasBlocker, and all was back to normal.  That one's the culprit.
